packages/astro/src/runtime/server/render/astro/render-template.ts

```diff
@@ -35,12 +36,23 @@ export class RenderTemplateResult {
 	async *[Symbol.asyncIterator]() {
 		const { htmlParts, expressions } = this;
 
+		let iterables: Array<EagerAsyncIterableIterator> = [];
+		// all async iterators start running in non-buffered mode to avoid useless caching
+		for (let i = 0; i < htmlParts.length; i++) {
+			iterables.push(new EagerAsyncIterableIterator(renderChild(expressions[i])));
+		}
+		// once the execution of the next for loop is suspended due to an async component,
+		// this timeout triggers and we start buffering the other iterators
+		setTimeout(() => {
```

Comment:
Why do this in a setTimeout and not immediately?

Reply:
@matthewp I guess that's the comment above https://github.com/withastro/astro/pull/7071#issuecomment-1549978562